Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.h

```diff
@@ -10,27 +10,52 @@
 
 #if !defined(Q_MOC_RUN)
 #include <AzCore/DOM/Backends/JSON/JsonBackend.h>
+#include <AzCore/Interface/Interface.h>
 #include <AzCore/Memory/SystemAllocator.h>
 #include <AzFramework/DocumentPropertyEditor/DocumentAdapter.h>
 #include <AzToolsFramework/UI/DocumentPropertyEditor/PropertyHandlerWidget.h>
-#include <AzCore/Interface/Interface.h>
-#include <QFrame>
 
-class QVBoxLayout;
-class QHBoxLayout;
+#include <QHBoxLayout>
+#include <QPointer>
+#include <QScrollArea>
 
 #endif // Q_MOC_RUN
 
 namespace AzToolsFramework
 {
     class DocumentPropertyEditor;
 
+    class DPELayout : public QHBoxLayout
+    {
+        // todo: look into caching and QLayoutItem::invalidate()
+    public:
+        DPELayout(int depth, QWidget* parentWidget = nullptr)
+            : QHBoxLayout(parentWidget)
+            , m_depth(depth)
+        {
+        }
+
+        // QLayout overrides
+        QSize sizeHint() const override;
+        QSize minimumSize() const override;
+        void setGeometry(const QRect& rect) override;
+        Qt::Orientations QLayoutItem::expandingDirections() const override
+        {
+            return Qt::Horizontal;
+        }
```

Comment:
Does this need to be defined inline, or can it go to the CPP?

Reply:
since I changed to basing off a QHBoxLayout instead a of a raw QLayout, it doesn't need to *be* at all. Removed.